flaml/tune/searcher/search_thread.py

```diff
@@ -25,6 +25,19 @@
 logger = logging.getLogger(__name__)
 
 
+def recursive_update(d:dict, u:dict):
+    """
+    Args:
+        d (dict): The target dictionary to be updated.
+        u (dict): A dictionary containing values to be merged into `d`.
```

Comment:
```suggestion
    Update the dictionary `d` with the dictionary `u` recursively.

    Args:
        d (dict): The target dictionary to be updated.
        u (dict): A dictionary containing values to be merged into `d`.

    Example:
    ```python
    d= {'num_boost_round': 10, 'params': {'max_depth': 12, 'eta': 0.02}}
    u= {'params': {'verbosity': 3, 'booster': 'gbtree'}}
    recursive_update(d, u)
    d  # {'num_boost_round': 10, 'params': {'max_depth': 12, 'eta': 0.02, 'verbosity': 3, 'booster': 'gbtree'}}
    ```
```